MetaMask/metamask-extension
# deprecate opensea security alerts check and settings

## Explanation

### Background
The new Blockaid integration should replace the existing OpenSea + Blockaid security provider feature. 

### Acceptance Criteria
1. Users that have the Opensea + Blockaid security provider enabled should have the new Blockaid feature enabled by default when it's rolled out
2. Users that have the Opensea + Blockaid security provider disabled should have the new Blockaid feature disabled by default when it's rolled out
3. The Opensea + Blockaid security provider should be disabled for all users - No data should be sent anymore to the API and no warning should be shown as a result of this feature
4. The Opensea + Blockaid security provider should not appear within settings anymore 
5. Metrics defined here https://github.com/MetaMask/MetaMask-planning/issues/527 should be triggered when users have the Blockaid feature enabled by default. (`blockaid` value should be added to the `security_providers` user property when it's enabled to users and Settings updated event should be triggered)
6. The `opensea` value should be removed from `security_providers` user property when it's disabled/deprecated for that user

### Out of scope 
We'll communicate users of this migration as part of the what's new text.

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/926

## Screenshots/Screencaps

### Before

**UI ScreenShot 1**
<img width="930" alt="Screenshot 2023-07-14 at 15 14 01" src="https://github.com/MetaMask/metamask-extension/assets/44811/38e49818-8255-4bd0-bd4f-65eca5c808c2">

**Transaction Data 1**
<img width="1256" alt="Screenshot 2023-07-28 at 11 40 51" src="https://github.com/MetaMask/metamask-extension/assets/44811/d0fea47f-5eed-41b9-8c17-d820b4083afb">

### After
**UI ScreenShot 2**
<img width="944" alt="Screenshot 2023-07-14 at 15 16 59" src="https://github.com/MetaMask/metamask-extension/assets/44811/587f01c9-76ee-42fb-865e-4be1ba9bf143">

**Transaction Data 2**
<img width="1262" alt="Screenshot 2023-07-28 at 11 43 17" src="https://github.com/MetaMask/metamask-extension/assets/44811/5d1453e3-7bcb-45bd-b64a-225baffd2931">

## Manual Testing Steps

1. Start MM using `yarn start`
2. Open Settings
3. Tab Experimental
4. You should see the screenshot like in **UI ScreenShot 1**
5. Send a transaction from test-dapp
6. Open console and type txData.securityProviderResponse, it should be null as in screenshot **Transaction Data 1**
7. Stop MM and restart using `yarn start:flask`
8. Open Settings
9. Tab Experimental
10. You should see the screenshot like in **UI ScreenShot 2**
11. Send a transaction from test-dapp
12. Open console and type txData.securityProviderResponse, it should be undefined as in screenshot **Transaction Data 2**


## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
